distribution/metadata/diff_metadata_service.go

```diff
@@ -0,0 +1,137 @@
+package metadata
+
+import (
+	"encoding/json"
+
+	"github.com/docker/distribution/digest"
+	"github.com/docker/docker/layer"
+)
+
+// DiffMetadataService maps layer IDs to a set of known metadata for
+// the layer.
+type DiffMetadataService struct {
```

Comment:
I hesitate to bring this up, because I know we've gone back and forth on naming here, and I don't want to hold up the PR over cosmetic naming issues. But I don't really like this name for a few reasons. First, the term "Diff" differs from the rest of the code base. Everywhere else we use the term "Layer". Every layer has a "DiffID". Also, "metadata" is a bit vague here. We have another service called `V1IDService`, which maps between `DiffIDs` and v1 registry IDs. To be consistent with that, we could call this service `V2MetadataService` or something similar.

Feel free to ignore if this is overly pedantic.